### PR TITLE
feat: Support Copilot metrics endpoints

### PR DESCRIPTION
**Pull request by henriklundstrom**
Introduces structs and functions for calling the newly GA endpoints for Copilot metrics. See this post from GitHub https://github.blog/changelog/2024-10-30-github-copilot-metrics-api-ga-release-now-available/.

### CODE REVIEW

File: github/copilot.go
Comment (claystation) [inline github/copilot.go:110]: Please add comments for each struct explaining the struct please.

File: github/copilot.go
Comment (henriklundstrom) [inline github/copilot.go:205]: According to the https://docs.github.com/en/rest/copilot/copilot-metrics?apiVersion=2022-11-28 response schema, these two are not required. Should I change them to be pointers to represent the case where they're omitted as nil?

Comment (gmlewis) [reply github/copilot.go:205]: Yes, please. Anything that is optional should be a pointer type with `omitempty`.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @henriklundstrom !
First, there are a few acronyms that need to be fixed for idiomatic Go naming conventions.

Then, before continuing my review, it seems like there is a lot of duplication in the new set of structs... are there structs (potential candidates are the ones utilizing `omitempty`) that could possibly be shared, or is there enough distinction to warrant the current number of structs?

I'm fine if the answer is the latter - I just wanted your opinion to see if you think we could take advantage of simplifying the interface by a bit of sharing (like we do with other structs). But again, if not, no big deal... then no changes necessary.

**Comment by henriklundstrom** [comment]
> Thank you, @henriklundstrom ! First, there are a few acronyms that need to be fixed for idiomatic Go naming conventions.
> 
> Then, before continuing my review, it seems like there is a lot of duplication in the new set of structs... are there structs (potential candidates are the ones utilizing `omitempty`) that could possibly be shared, or is there enough distinction to warrant the current number of structs?
> 
> I'm fine if the answer is the latter - I just wanted your opinion to see if you think we could take advantage of simplifying the interface by a bit of sharing (like we do with other structs). But again, if not, no big deal... then no changes necessary.

@gmlewis I've gone through your renaming suggestions and applied them all.

I agree there are many structs being introduced and that they're somewhat repetitive, but still not actually modular, and it's not a good read... Unfortunately this is how GitHub has structured the response schema. In this case I don't see that using struct embedding or reusable structs will make it any better. Possibly, I could introduce some:
```
type CopilotModelBase struct {
	Name                    string  `json:"name"`
	IsCustomModel           bool    `json:"is_custom_model"`
	CustomModelTrainingDate *string `json:"custom_model_training_date,omitempty"`
}
```
That would cut down the size of the `CopilotIDECodeCompletionsModel`, `CopilotIDEChatModel`, `CopilotDotcomChatModel`, and `CopilotDotcomPullRequestsModel` structs, but not by much and I'm not sure it makes them any nicer to work with.

**Comment by gmlewis** [comment]
@henriklundstrom - to proceed with this PR, it looks like you need to do the following steps:

* `git checkout master`
* `git pull upstream master` (or however you set up your repo to pull from the upstream repo)
* `git checkout nordnet:copilot-metrics`
* `git merge master`
* `./script/generate.sh` (see step 4 of CONTRIBUTING.md for more details)
* `git push origin nordnet:copilot-metrics`

This should pull #3352 into your PR which has the latest OpenAPI changes needed to support `copilot-metrics`.

**Review by gmlewis — APPROVED** [review]
Thank you, @henriklundstrom !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by claystation — APPROVED** [review]
LGTM :)

**Comment by henriklundstrom** [comment]
@gmlewis I've updated the branch and we have a second approval now 🚀

**Comment by gmlewis** [comment]
Thank you, @claystation !
Merging.

**Pull request merged**
